sync/sync-store/src/test/java/com/duckduckgo/sync/store/TestSharedPrefsProvider.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.sync.store
+
+import android.content.Context
+import android.content.SharedPreferences
+
+class TestSharedPrefsProvider(val context: Context) : SharedPrefsProvider {
+    override fun getSharedPrefs(fileName: String): SharedPreferences? {
+        return context.getSharedPreferences(fileName, Context.MODE_PRIVATE)
+    }
+}
```

Comment:
Probably you need this class nor doing a robolectric test. In `common-test` module we have `InMemorySharedPreferences` that can be use to do JVM test that require a shared prefs.

Reply:
I guess my point is why we need to use robolectric at all when we have an in-memory shared prefs we can use (?)